### PR TITLE
adding pnp

### PR DESCRIPTION
**Pull request by edgarriba**

### CODE REVIEW

**Comment by vrabaud** [comment]
Please fix until Travis passes. Thx.

File: src/training/Trainer.cpp [hunk 1]
```diff
@@ -101,54 +108,115 @@ struct Trainer {
...
+
       // TODO actually use the params and do not force ORB
 #if CV_VERSION_MAJOR == 3
-      cv::Ptr<cv::DescriptorExtractor> orb = cv::ORB::create();
-      orb->detectAndCompute(obs.image, obs.mask, keypoints, descriptors);
+      feature2d_->detectAndCompute(obs.image, obs.mask, keypoints, descriptors);
 #else
-      cv::ORB orb;
-      orb(obs.image, obs.mask, keypoints, descriptors);
+      (*feature2d_)(obs.image, obs.mask, keypoints, descriptors);
```
Comment (edgarriba) [inline src/training/Trainer.cpp]: @vrabaud not sure if all opencv2 features2d have this signature

Comment (edgarriba) [inline src/training/Trainer.cpp]: Besides, I think it's not coherent that in detection we use ecto_opencv in order to compute the features and then at training time just call the OpenCV C++ API.

Comment (vrabaud) [inline src/training/Trainer.cpp]: I think there is no good reason :)

**Comment by vrabaud** [comment]
Please squash your commits and we are good to go. Thx ! That was a big one.

**Comment by edgarriba** [comment]
@vrabaud something wrong happened with Travis

**Comment by linknum23** [comment]
@edgarriba I just tested this using your detection.ros.ork and it seems to work. My test was getting very few inliers using the standard (non PNP) mode. It gets many more inliers using PNP mode. 
I noticed that the detection code in PNP mode still requires a depth image, since it stalls waiting for one. Is there a reason for that? I assumed with PNP mode a depth image would not be required...

**Comment by edgarriba** [comment]
@linknum23 you are totally right. No depth image is needed. Once this PR is merge we can start refactoring this things.

@vrabaud please merge

**Pull request closed without merging**